packages/mcdoc/src/runtime/attribute/builtin.ts

```diff
@@ -277,4 +290,35 @@ export function registerBuiltinAttributes(meta: core.MetaRegistry) {
 			}
 		},
 	})
+	registerAttribute(meta, 'match_regex', regexMatchValidator, {
+		checker: (config, typeDef, _) => {
+			if (typeDef.kind !== 'literal' || typeDef.value.kind !== 'string') {
+				return undefined
+			}
+			const pattern = config.regex
+			const value = typeDef.value.value
+			return (node, ctx) => {
+				try {
+					const regex = RegExp(pattern)
+					if (!regex.test(value)) {
+						if (config.error) {
+							ctx.err.report(config.error, node, 2)
+						} else {
+							ctx.err.report(
+								localize('mismatching-regex-pattern', typeDef.value.value),
+								node,
+								2,
```

Comment:
Use `core.ErrorSeverity.Warning` instead of `2` here. I know this file already used `2` for this but that was probably a mistake.